pkg/reader/config_poller.go

```diff
@@ -65,11 +67,333 @@ func newConfigPoller(
 	refreshPeriod time.Duration,
 ) *configPoller {
 	return &configPoller{
-		chainCaches:   make(map[cciptypes.ChainSelector]*chainCache),
-		refreshPeriod: refreshPeriod,
-		reader:        reader,
-		lggr:          lggr,
+		chainCaches:       make(map[cciptypes.ChainSelector]*chainCache),
+		refreshPeriod:     refreshPeriod,
+		reader:            reader,
+		lggr:              lggr,
+		knownSourceChains: make(map[cciptypes.ChainSelector]map[cciptypes.ChainSelector]bool),
+		stopChan:          make(chan struct{}),
+	}
+}
+
+// startBackgroundPolling starts a goroutine that periodically refreshes the cache
+func (c *configPoller) startBackgroundPolling() {
+	c.wg.Add(1)
+	go func() {
+		defer c.wg.Done()
+		ticker := time.NewTicker(c.refreshPeriod)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ticker.C:
+				c.refreshAllKnownChains()
+			case <-c.stopChan:
+				return
+			}
+		}
+	}()
+}
+
+// Add Start method to configPoller
+func (c *configPoller) Start() {
+	c.startBackgroundPolling()
+
+	c.lggr.Info("Background poller started")
+}
+
+// Stop stops the background polling
+func (c *configPoller) Stop() {
+	close(c.stopChan)
+	c.wg.Wait()
+}
+
+// refreshAllKnownChains refreshes all known chains in background using batched requests where possible
+func (c *configPoller) refreshAllKnownChains() {
+	c.RLock()
+	// Gather all chain configs and source chain configs to refresh
+	chainSelectors := make([]cciptypes.ChainSelector, 0, len(c.chainCaches))
+	sourceChainsMap := make(map[cciptypes.ChainSelector][]cciptypes.ChainSelector)
+
+	for chainSel := range c.chainCaches {
+		chainSelectors = append(chainSelectors, chainSel)
+
+		// If this chain has source chains, gather them
+		if sourceChains, exists := c.knownSourceChains[chainSel]; exists && len(sourceChains) > 0 {
+			sourceList := make([]cciptypes.ChainSelector, 0, len(sourceChains))
+			for sourceChain := range sourceChains {
+				sourceList = append(sourceList, sourceChain)
+			}
+			sourceChainsMap[chainSel] = sourceList
+		}
+	}
+	c.RUnlock()
+
+	// Refresh each chain (and its source chains if applicable)
+	for _, destChain := range chainSelectors {
+		// Skip chains we don't know as dest chains
+		sourceChains, hasSourceChains := sourceChainsMap[destChain]
+
+		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
```

Comment:
nit: `30 * time.Second` can be a const `bgRefreshTimeout` at the top of this file